lib/chef/provider/aws_ebs_volume.rb

```diff
@@ -204,4 +204,14 @@ def delete(volume)
       volume
     end
   end
+
+  def availability_zone
```

Comment:
True, but this maintains the previous functionality, and we didn't validate for that before letting EC2 catch that specific exception.  You're code looks nicer anyhow, so I'll make the change.